src/main/java/edu/ucsd/workflow/ShuffleFASTA.java

```diff
@@ -45,27 +45,42 @@ public static void main(String[] args) {
 			// read and reverse sequences one by one from input file
 			String line = null;
 			StringBuffer sequence = null;
+			// If sequence is already reversed, don't write the reversed
+			// version of that
+			boolean prevProteinIsDecoy = false;
 			while (true) {
 				line = input.readLine();
 				if (line == null) {
 					// if a previous sequence is still being processed,
 					// reverse it and write it to the output file
-					if (sequence != null && sequence.length() > 0)
+					if (
+						sequence != null && sequence.length() > 0 &&
+						(!prevProteinIsDecoy)
+					) {
 						writeReversedSequence(sequence, output);
+					}
 					break;
 				} else if (line.startsWith(">")) {
 					// process the FASTA comment line to
 					// insert the proper decoy prefix
 					line = line.substring(1).trim();
-					line = ">" + DECOY_PREFIX + "_" + line;
 					// if a previous sequence is still being processed,
 					// reverse it and write it to the output file
-					if (sequence != null && sequence.length() > 0)
+					if (
+						sequence != null && sequence.length() > 0 &&
+						(!prevProteinIsDecoy)
+

```

Comment:
Why exactly is there a blank line here?

Reply:
The line break is there to conform to the max line length for this script (which seems to be 80 characters).

Reply:
I'm not referring to the line break between lines 70 and 71. That's fine. I'm referring to line 72, which is completely blank.

Reply:
Removed the blank line